### PR TITLE
readelf is Linux only

### PR DESCRIPTION
**Pull request by rex4539**
Fixes https://github.com/zcash/zcash/issues/4875.

### CODE REVIEW

**Review by str4d — APPROVED** [review]

**Comment by str4d** [comment]
@zkbot r+

**Comment by daira** [comment]
Looks like:
* `check-security` is failing on many builders
* native_ccache 4.1 needs to be postponed.

File: configure.ac [hunk 1]
```diff
@@ -106,7 +106,12 @@ AC_PATH_PROG(CCACHE,ccache)
 dnl This one is not currently used anywhere, thus not required:
 AC_PATH_PROG(XGETTEXT,xgettext)
 ZC_REQUIRE_PROG(HEXDUMP,hexdump)
-ZC_REQUIRE_TOOL(READELF,readelf)
+
+dnl readelf is Linux only
+if test x$TARGET_OS = xlinux; then
+  ZC_REQUIRE_TOOL(READELF,readelf)
+fi
+
```
Comment (str4d) [inline configure.ac:114]: I think the problem here is that this does not revert the original change, so now nothing is configured for non-Linux platforms. Let's try this instead:
```suggestion
AC_PATH_PROG(READELF,readelf)
```

Comment (rex4539) [reply configure.ac:114]: Hm, my change compiles fine on macOS. I don't think the CI failures were related to this commit. But I can try your proposal when I'm back at my computer.

Comment (rex4539) [reply configure.ac:114]: I tested your suggestion and it still doesn't affect macOS, so in that sense, bring it in!

Comment (rex4539) [reply configure.ac:114]: Your suggestion works for macOS but I can't vouch for other platforms so I'll take your word for it :)

**Review by daira — APPROVED** [review]
utACK

**Comment by daira** [comment]
@zkbot r+

**Pull request merged**